IPython/terminal/interactiveshell.py

```diff
@@ -1,21 +1,589 @@
+                            ).tag(config=True)
+
+    def _space_for_menu_changed(self, old, new):
+        self._update_layout()
+
+    pt_cli = None
+    debugger_history = None
+
+    simple_prompt = Bool(_use_simple_prompt,
+        help="""Use `raw_input` for the REPL, without completion, multiline input, and prompt colors.
+
+            Useful when controlling IPython as a subprocess, and piping STDIN/OUT/ERR. Known usage are:
+            IPython own testing machinery, and emacs inferior-shell integration through elpy.
+
+            This mode default to `True` if the `IPY_TEST_SIMPLE_PROMPT`
+            environment variable is set, or the current terminal is not a tty.
+
+            """
+            ).tag(config=True)
+
+    @property
+    def debugger_cls(self):
+        return Pdb if self.simple_prompt else TerminalPdb
+
+    autoedit_syntax = Bool(False,
+        help="auto editing of files with syntax errors.",
+    ).tag(config=True)
+
+
+    confirm_exit = Bool(True,
+        help="""
+        Set to confirm when you try to exit IPython with an EOF (Control-D
+        in Unix, Control-Z/Enter in Windows). By typing 'exit' or 'quit',
+        you can force a direct exit without any confirmation.""",
+    ).tag(config=True)
+
+    editing_mode = Unicode('emacs',
+        help="Shortcut style to use at the prompt. 'vi' or 'emacs'.",
+    ).tag(config=True)
+
+    mouse_support = Bool(False,
+        help="Enable mouse support in the prompt"
+    ).tag(config=True)
+
+    highlighting_style = Unicode('default',
+            help="The name of a Pygments style to use for syntax highlighting: \n %s" % ', '.join(get_all_styles())
+    ).tag(config=True)
+
+    
+    @observe('highlighting_style')
+    def _highlighting_style_changed(self, change):
+        self._style = self._make_style_from_name(self.highlighting_style)
+
+    highlighting_style_overrides = Dict(
+        help="Override highlighting format for specific tokens"
+    ).tag(config=True)
+
+    editor = Unicode(get_default_editor(),
+        help="Set the editor used by IPython (default to $EDITOR/vi/notepad)."
+    ).tag(config=True)
+
+    prompts_class = Type(Prompts, help='Class used to generate Prompt token for prompt_toolkit').tag(config=True)
+
+    prompts = Instance(Prompts)
+
+    @default('prompts')
+    def _prompts_default(self):
+        return self.prompts_class(self)
+
+    @observe('prompts')
+    def _(self, change):
+        self._update_layout()
+
+    @default('displayhook_class')
+    def _displayhook_class_default(self):
+        return RichPromptDisplayHook
+
+    term_title = Bool(True,
+        help="Automatically set the terminal title"
+    ).tag(config=True)
+
+    display_completions_in_columns = Bool(None,
+        help="Display a multi column completion menu.", allow_none=True
+    ).tag(config=True)
+
+    @observe('display_completions_in_columns')
+    def _display_completions_in_columns_changed(self, new):
```

Comment:
No, i'll remove it post beta. I know of a few person that have set up this
option. It's (relatively) cheap to keep.
On Jun 16, 2016 04:34, "Min RK" <notifications@github.com> wrote:

> In IPython/terminal/interactiveshell.py
> <https://github.com/ipython/ipython/pull/9627#discussion_r67329314>:
>
> > +        self._update_layout()
> > +
> > +    @default('displayhook_class')
> > +    def _displayhook_class_default(self):
> > +        return RichPromptDisplayHook
> > +
> > +    term_title = Bool(True,
> > +        help="Automatically set the terminal title"
> > +    ).tag(config=True)
> > +
> > +    display_completions_in_columns = Bool(None,
> > +        help="Display a multi column completion menu.", allow_none=True
> > +    ).tag(config=True)
> > +
> > +    @observe('display_completions_in_columns')
> > +    def _display_completions_in_columns_changed(self, new):
>
> Do we need a deprecation for an unreleased feature that's only been in
> master briefly?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipython/ipython/pull/9627/files/0668587b74996a0c1bb043687902c5d0738e177a#r67329314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAUezyc750QW_TE8LTLZHC7ViHWf5ou7ks5qMTTegaJpZM4I2349>
> .
>